### PR TITLE
Update RootlessKit to v2.0.0 (support Pasta)

### PR DESCRIPTION
**Pull request by AkihiroSuda**
https://github.com/rootless-containers/rootlesskit/releases/tag/v2.0.0

## Pasta
RootlessKit v2 adds the support for pasta (https://passt.top/passt/).
Pasta is similar to slirp4netns but its port forwarder achieves better throughput than slirp4netns port driver.

It is still not faster than RootlessKit's `builtin` port driver, but unlike the `builtin` port driver, pasta can retain source IP address information.

Network driver | Port driver    | Net throughput | Port throughput | Src IP | No SUID | Note
---------------|----------------|----------------|-----------------|--------|---------|--------------------------------------------
slirp4netns    | builtin        | Slow           | Fast ✅         | ❌     | ✅      | Default in typical setup
vpnkit         | builtin        | Slow           | Fast ✅         | ❌     | ✅      | Default when slirp4netns is not installed
slirp4netns    | slirp4netns    | Slow           | Slow            | ✅     | ✅      |
**pasta**      | **implicit**   | Slow           | Fast ✅         | ✅     | ✅      | Experimental; Needs recent version of pasta (`2023_12_04.b86afe3`)
lxc-user-nic   | builtin        | Fast ✅        | Fast ✅            | ❌     | ❌      | Experimental
(bypass4netns) | (bypass4netns) | Fast ✅        | Fast ✅         | ✅     | ✅      | (Not integrated to RootlessKit; needs another daemon and a custom seccomp profile)

## Detach-netns
Aside from pasta, RootlessKit v2 also brings the support for "detach-netns" mode, which leaves the runtime in the host network namespace to eliminate the slirp overhead for pull/push and to allow accessing the "real" 127.0.0.1.

See containerd/nerdctl#2723 for how detach-netns is being adopted in nerdctl v2.

Integrating detach-netns into Docker/Moby will need an extra work and will be deferred to Docker v26 (or later).

- - -

**- What I did**
Update RootlessKit to v2.0.0.

**- How I did it**
Update RootlessKit to v2.0.0.

**- How to verify it**

- No visible change for the default configuration
- To enable pasta:
  -  Install pasta (https://passt.top/passt/) `2023_12_04.b86afe3` or later
  - Set `DOCKERD_ROOTLESS_ROOTLESSKIT_NET=pasta` and `DOCKERD_ROOTLESS_ROOTLESSKIT_PORT_DRIVER=implicit` via `~/.config/systemd/user/docker.service.d/override.conf`
  - Restart the daemon (`systemctl --user daemon-reload && systemctl --user restart docker`)
  - Confirm that `pasta` process is running

**- Description for the changelog**

Update RootlessKit to v2.0.0. (support Pasta)

**- A picture of a cute animal (not mandatory but encouraged)**
🐧

### CODE REVIEW

File: vendor.mod [hunk 1]
```diff
@@ -46,8 +46,8 @@ require (
 	github.com/gogo/protobuf v1.3.2
 	github.com/golang/gddo v0.0.0-20190904175337-72a348e765d2
 	github.com/golang/protobuf v1.5.3
-	github.com/google/go-cmp v0.5.9
-	github.com/google/uuid v1.3.0
+	github.com/google/go-cmp v0.6.0
+	github.com/google/uuid v1.5.0
```
Comment (thaJeztah) [inline vendor.mod:50]: Oh! I recall I had a PR to update these, but at the time we decided it wasn't critical for 25.0, so moved it to draft; I just did a quick rebase, and can move it out of draft if we need this one;

- https://github.com/moby/moby/pull/46980

File: vendor.mod [hunk 2]
```diff
@@ -67,7 +67,7 @@ require (
 	github.com/moby/pubsub v1.0.0
 	github.com/moby/swarmkit/v2 v2.0.0-20230911190601-f082dd7a0cee
 	github.com/moby/sys/mount v0.3.3
-	github.com/moby/sys/mountinfo v0.6.2
+	github.com/moby/sys/mountinfo v0.7.1
```
Comment (thaJeztah) [inline vendor.mod:70]: Maybe we should update this one (and the `x/sys` one separately; in case we need to update those in release branches as well).

**Comment by thaJeztah** [comment]
Other dependencies were updated in separate PRs; could you do a rebase of this one to remove them from the diff?

**Comment by AkihiroSuda** [comment]
> Other dependencies were updated in separate PRs; could you do a rebase of this one to remove them from the diff?

Thanks, rebased

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by vvoland — APPROVED** [review]

**Comment by thaJeztah** [comment]
Let's bring this one in 👍

File: contrib/dockerd-rootless.sh [hunk 3]
```diff
@@ -9,12 +9,31 @@
...
+
+# Guide to choose the network driver and the port driver:
 #
+#  Network driver | Port driver    | Net throughput | Port throughput | Src IP | No SUID | Note
+#  ---------------|----------------|----------------|-----------------|--------|---------|---------------------------------------------------------
+#  slirp4netns    | builtin        | Slow           | Fast ✅         | ❌     | ✅      | Default in typical setup
+#  vpnkit         | builtin        | Slow           | Fast ✅         | ❌     | ✅      | Default when slirp4netns is not installed
+#  slirp4netns    | slirp4netns    | Slow           | Slow            | ✅     | ✅      |
+#  pasta          | implicit       | Slow           | Fast ✅         | ✅     | ✅      | Experimental; Needs recent version of pasta (2023_12_04)
+#  lxc-user-nic   | builtin        | Fast ✅        | Slow            | ❌     | ❌      | Experimental
```
Comment (AkihiroSuda) [inline contrib/dockerd-rootless.sh:34]: Sorry, "Slow" here had to be "Fast"

Comment (AkihiroSuda) [reply contrib/dockerd-rootless.sh:34]: - https://github.com/moby/moby/pull/47083

**Pull request merged**